compiler/quilt/test/test_cli.py

```diff
@@ -0,0 +1,589 @@
+    combination of arguments that has been created, and/or that tests
+    for arguments that no longer exist should be removed (or verified).
+
+    This helps to recognize when a change has been made that might be a
+    breaking change.
+
+    Key paths take the form:
+        [0, 'version', 0, 'add', 0]
+
+
+=== CLI <--> Python API Testing ===
+
+To test that the CLI is calling the API correctly, we mock the
+command module using the MockObject class.  During this process,
+the following steps occur:
+    * params for the specific commands to be tested are marked as
+      tested in TESTED_PARAMS.
+    * If the OS environment has QUILT_TEST_CLI_SUBPROC=True
+      * Quilt is called as a subprocess with the specified options
+        and QUILT_TEST_CLI_SUBPROC="True" in the environment
+      * QUILT_TEST_CLI_SUBPROC is read and MockObject is used
+        by tools/main.py
+      * MockObject emits JSON to stdout
+      * JSON is read from subprocess
+    * Otherwise:
+      * quilt.tools.main.command is mocked with MockObject
+      * quilt.tools.main.argument_parser() is called
+      * quilt.tools.main.main() is called with parsed args
+    * Function calls are read from MockObject instance
+    * Function is verified to exist in actual command module
+    * args and kwargs are bound to the function (without calling it)
+      to ensure that the arguments work.
+    * Types, values, or other conditions to be sent to the command
+      function may be checked by the specific test as well.
+
+The binding step should prevent any CLI parameters from changes or
+removal (or incompatible additions) on the API side.
+"""
+#NOTE: This is slow in subproc mode due to frequent quilt executions (see #194)
+#TODO: More tests!
+#TODO: remove xfail from test_coverage() once coverage is complete
+
+# BUG: (minor) add args/kwargs items into param trees.
+# It's not very easy to do this, but it fixes the issue that
+# we currently leave some ambiguity as to whether [0, 'foo', 'bar']
+# indicates:
+#     Anonymous positional argument 0
+#     Followed by required argument 'foo'
+#     Followed by required or optional argument 'bar'
+# or indicates:
+#     Positional 0, when 'foo' is given
+#     Followed by required or optional argument "bar"
+#
+# This means if the argument order is restructured in a particular way,
+# a change may not be noticed.  And, as we all know, if it's possible,
+# it will eventually occur.
+# Fixing this improves our coverage, but might be somewhat complex.
+
+import os
+import sys
+import json
+import inspect
+import collections
+from subprocess import call, check_output, PIPE, CalledProcessError
+
+import pytest
+
+from .utils import BasicQuiltTestCase
+
+# inspect.argspec is deprecated, so
+try:
+    from funcsigs import signature  # python 2.7
+except ImportError:
+    from inspect import signature
+
+
+## "Static" vars
+_TEST_DIR = os.path.split(os.path.abspath(__file__))[0]
```

Comment:
Use `os.path.dirname`

Reply:
Original code:

@dimaryaz wrote:
> There's os.path.split (which also supports both / and \ on win32)

Was told by you to use os.path.split, over splitting on os.path.sep (which is /, or \ on win32).
Changed to your current preference, and changed other occurrences as well.